clients/lite-xl/upstream/init.lua

```
@@ -719,118 +719,162 @@ local function apply_edit(server, doc, text_edit, is_snippet, update_cursor_posi
-    local admitted, disposition = lsp.admit_response(item.data.subject)
-    if not admitted then
+---Apply the selected item exactly once from its final effective fields
+---(#11188). Resolution outcomes decide the effective item: a resolved item
+---overlays the original; a not_needed item applies as received; failed,
+---timed_out, and stale terminals fall back only when the original proves its
+---own complete application surface (bytes plus known-empty-or-present
+---additional edits) and otherwise refuse without partial mutation.
+local function apply_selected_completion(item, rstate)
+  if rstate.applied then return true end
+  local original = item.data.completion_item
+  local effective = nil
```

Luacheck (1.2.0)

[warning] value assigned to variable 'effective' is unused

(W311)